logbook/handlers.py

```diff
@@ -1391,14 +1391,26 @@ def __init__(self, application_name=None, address=None,
             else:
                 address = '/dev/log'
 
-        self.address = address
+        self.remote_address = self.address = address
         self.facility = facility
         self.socktype = socktype
 
         if isinstance(address, string_types):
             self._connect_unixsocket()
+            self.enveloper = self.unix_envelope
+            default_delimiter = u'\x00'
         else:
             self._connect_netsocket()
+            self.enveloper = self.net_envelope
+            default_delimiter = u'\n'
+
+        self.record_delimiter = default_delimiter \
+            if record_delimiter is None else record_delimiter
+
+        try:
+            self.connection_exception = BrokenPipeError
+        except NameError:
```

Comment:
This is a dangerous way of checking if BrokenPipeError is supported.

Reply:
Why? I don't see what's dangerous about it.

Can you suggest an alternative that is as portable?

Reply:
Catching `NameError` is considered dangerous because this would also raise it:

```
sellf.connection_exception = BrokenPipeError
```
And also this (which is a common typo for some folks):
```
shelf.connection_exception = ...
```
Also, at some point in the future `self` could implement `__setattr__` or `__setattribute__` which themselves could introduce the exception, and this would quietly swallow it.

A better way is to use
```
getattr(__builtin__, 'BrokenPipeError', <fallback>)
```